src/Data/Path/Pathy.purs

```diff
@@ -478,6 +477,27 @@ module Data.Path.Pathy
   parseAbsDir :: String -> Maybe (AbsDir Unsandboxed)
   parseAbsDir = parsePath (const Nothing) (const Nothing) (const Nothing) Just
 
+  -- | Folds a `Path` into a value using a specified function and a provided
+  -- | default.
+  fold :: forall c a b s. (String -> c -> c) -> c -> Path a b s -> c
+  fold _ z (Current                 ) = z
+  fold _ z (Root                    ) = z
+  fold f z (ParentIn p              ) = fold f z p
+  fold f z (FileIn   p (FileName f')) = f f' (fold f z p)
+  fold f z (DirIn    p (DirName  f')) = f f' (fold f z p)
```

Comment:
That's a bit nitpickish, but I feel confused when something of type `String` is called `f'`, it probably would be better to call it `a`, but that's not very important.